tests/test_TFNativeOp.py

```diff
@@ -3033,6 +3033,95 @@ def test_blocksparse_simple_feature_axis1():
+                   0.1, 0.7, 0.1, 0.2, 0.1, 0.1], dtype=np.float32).reshape(1, 2, 3, 5)
+
+  labels = np.array([[1, 2]], dtype=np.int32)
+  input_lengths = np.array([2], dtype=np.int32)
+  label_lengths = np.array([2], dtype=np.int32)
+
+  acts_t = tf.constant(acts)
+  labels_t = tf.constant(labels)
+  input_lengths_t = tf.constant(input_lengths)
+  label_lengths_t = tf.constant(label_lengths)
+  acts_t = tf.nn.log_softmax(acts_t)  # NOTE cpu
+  costs = rnnt_loss(acts_t, labels_t, input_lengths_t, label_lengths_t)
+  print(costs.eval())
+
+
+@unittest.skipIf(not is_gpu_available(), "no gpu on this system")
+def test_warprnnt_multiple_batches():
+  import numpy as np
+  n_batch = 2
+  n_time = 4
+  n_target = 3
+  n_vocab = 3
+
+  acts = np.array([0.065357, 0.787530, 0.081592, 0.529716, 0.750675, 0.754135,
+                   0.609764, 0.868140, 0.622532, 0.668522, 0.858039, 0.164539,
+                   0.989780, 0.944298, 0.603168, 0.946783, 0.666203, 0.286882,
+                   0.094184, 0.366674, 0.736168, 0.166680, 0.714154, 0.399400,
+                   0.535982, 0.291821, 0.612642, 0.324241, 0.800764, 0.524106,
+                   0.779195, 0.183314, 0.113745, 0.240222, 0.339470, 0.134160,
+                   0.505562, 0.051597, 0.640290, 0.430733, 0.829473, 0.177467,
+                   0.320700, 0.042883, 0.302803, 0.675178, 0.569537, 0.558474,
+                   0.083132, 0.060165, 0.107958, 0.748615, 0.943918, 0.486356,
+                   0.418199, 0.652408, 0.024243, 0.134582, 0.366342, 0.295830,
+                   0.923670, 0.689929, 0.741898, 0.250005, 0.603430, 0.987289,
+                   0.592606, 0.884672, 0.543450, 0.660770, 0.377128, 0.358021], dtype=np.float32)
+  acts = np.reshape(acts, (n_batch, n_time, n_target, n_vocab))
+
+  expected_costs = np.array([4.28065, 3.93844], dtype=np.float32)
+  expected_grads = np.array([-0.186844, -0.062555, 0.249399, -0.203377, 0.202399, 0.000977,
+                             -0.141016, 0.079123, 0.061893, -0.011552, -0.081280, 0.092832,
+                             -0.154257, 0.229433, -0.075176, -0.246593, 0.146405, 0.100188,
+                             -0.012918, -0.061593, 0.074512, -0.055986, 0.219831, -0.163845,
+                             -0.497627, 0.209240, 0.288387, 0.013605, -0.030220, 0.016615,
+                             0.113925, 0.062781, -0.176706, -0.667078, 0.367659, 0.299419,
+                             -0.356344, -0.055347, 0.411691, -0.096922, 0.029459, 0.067463,
+                             -0.063518, 0.027654, 0.035863, -0.154499, -0.073942, 0.228441,
+                             -0.166790, -0.000088, 0.166878, -0.172370, 0.105565, 0.066804,
+                             0.023875, -0.118256, 0.094381, -0.104707, -0.108934, 0.213642,
+                             -0.369844, 0.180118, 0.189726, 0.025714, -0.079462, 0.053748,
+                             0.122328, -0.238789, 0.116460, -0.598687, 0.302203, 0.296484], dtype=np.float32)
```

Comment:
How did you calculate this?

Reply:
I just used their test cases. https://github.com/HawkAaron/warp-transducer/blob/master/tensorflow_binding/tests/test_warprnnt_op.py

Should I calculate it myself to verify? (e.g. using a reference implementation)

Reply:
For this first pull request, that's ok (but put this link as a comment to the code then).
As a next step, it would be good to have such a reference implementation (also flexible to allow for other cases which we want to study later).